src/useOverlay.tsx

```diff
@@ -0,0 +1,68 @@
+import { RefObject, useEffect } from 'react';
+
+export function useOverlay(
+  rootElementId: string,
+  isOpen: boolean,
+  close: () => void,
+  ref: RefObject<HTMLDivElement>
+): void {
+  useEffect(() => {
+    if (!isOpen) {
+      return;
+    }
+
+    const handleKeydown = (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        close();
+        return;
+      }
+
+      if (event.key === 'Tab') {
+        event.preventDefault();
+
+        if (ref.current === null) {
+          return;
+        }
+
+        // https://gomakethings.com/how-to-get-the-first-and-last-focusable-elements-in-the-dom/
+        const focusables: HTMLElement[] = Array.from(
+          ref.current.querySelectorAll(
+            'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])'
+          )
+        );
+
+        const currentIndex = focusables.findIndex(
+          (element) => element === document.activeElement
+        );
+
+        if (currentIndex === -1) {
+          focusables[0].focus();
+          return;
+        }
+
+        const nextIndex = currentIndex + (event.shiftKey ? -1 : 1);
+
+        if (nextIndex === -1) {
+          focusables[focusables.length - 1].focus();
+          return;
+        }
+
+        if (nextIndex === focusables.length) {
+          focusables[0].focus();
+          return;
+        }
+
+        focusables[nextIndex].focus();
+        return;
+      }
+    };
+
+    document.getElementById(rootElementId)?.setAttribute('aria-hidden', 'true');
+    document.body.addEventListener('keydown', handleKeydown);
+
+    return () => {
+      document.getElementById(rootElementId)?.removeAttribute('aria-hidden');
+      document.body.removeEventListener('keydown', handleKeydown);
+    };
```

Comment:
```suggestion
    document.body.addEventListener('keydown', handleKeydown);

    return () => {
      document.body.removeEventListener('keydown', handleKeydown);
    };
```

Since `aria-hidden="true"` will be added to the entire root to be mounted, and the modal itself will be hidden, this description should not be used for accessibility reasons.

Reply:
Ok, I'll remove it and remove `rootElementId` from the arguments of this hook.